### ISSUE TITLE
Homepage GitHub bubble borks if seen more than 30 times in one hour

### ISSUE THREAD

**Issue by marcuswhybrow** [marcuswhybrow/ray-peat-rodeo#57]
The homepage has a bubble that looks like this:

![image](https://github.com/marcuswhybrow/ray-peat-rodeo/assets/278456/1673cd1d-96c8-49db-bcac-7175f5b4adf4)

It's data is fetched in the client JavaScript via two unauthenticated requests to the GitHub REST API. This API is rate limited to 60 requests per hour, equating to 30 refreshes per hour max. And this doesn't consider other API use from the same IP address which counts towards rate limiting, e.g. VPN's share IPs address between multiple users (I think). Hitting the rate limit looks like this:

![image](https://github.com/marcuswhybrow/ray-peat-rodeo/assets/278456/f3efa8e9-0b33-43ca-8a01-a1e9495a473f)

Graceful fallback needed methinks. However this might be confusing: "where has the commit message gone, I just saw it." Annoying too would be a failure message "API rate limit exceeded" or something more kindly worded. Can you suggest to "come back later" perhaps, but VPN users will likely never get any results from the GitHub API.

**Comment by marcuswhybrow** [comment]
I wonder if GitHub actions could get the same information at the deployment stage, and output the data to `./build/data/latest-commit.json`. Still use JS to async fetch the data into the page, that keeps the integration simple. 

It may not be immediately obvious, but you can't have the commit data inside the codebase itself, because that data is generated (hash/message/etc.) only after any code is committed. 

However, maybe one can pull data in at build time from git, not just during a GitHub action deployment, but more portably during every build. There must be Go bindings for git to scrape the data.

**Comment by marcuswhybrow** [comment]
[git2go](https://github.com/libgit2/git2go) looks promissing.

**Comment by marcuswhybrow** [comment]
e6a6b9c makes this a non-issue.

![image](https://github.com/marcuswhybrow/ray-peat-rodeo/assets/278456/4701ff34-3913-4fa1-a1d9-9a6cfcceec6d)

- Now the latest commit data is determined at build time by interrigating the git repository. (Annotated in orange)
- That's fine because every new commit triggers a rebuild and redeployment of the website to GitHub pages.
- However, the stars, forks, and issues can change between commits (circled in green in the image above).
- These metrics are still fetched at runtime using JavaScript.
- API rate limiting happens half as often, since API called are reduced from 2 to 1, total.
- Hardoded ballpark metrics are displayed if the GitHub API fails dur to rate limiting.

So I've calling this done.

**Comment by marcuswhybrow** [comment]
Ok, so e6a6b9c [broke the build](https://github.com/marcuswhybrow/ray-peat-rodeo/actions/runs/8784442625/job/24102679581), and as I was fixing the immediate mistake I made I realised it cannot work in principle.

The mistake I had made was to not tell `nix` to have `libgit2` and `pkg-config` in the path at build time, both of which are needed by `git2go` golang module. 

I fixed that, but it still fails: there is no git repo present in the source files that nix sees when building the derivation. i.e. `.git` doesn't exist. Nix prevents you from doing the kind of thing I'm trying to do. Which makes sense, because it's kind of impure: the git data is not part of the source code.

If found [Accessing .git directory in flake with local source](https://discourse.nixos.org/t/accessing-git-directory-in-flake-with-local-source/17370/1) on the Nix forums.

**Comment by marcuswhybrow** [comment]
Back to using two unauthenticated GitHub REST API calls with 465b5a3. 

At least API failure now looks like this:

![image](https://github.com/marcuswhybrow/ray-peat-rodeo/assets/278456/de1335cf-f725-4938-a76e-6424aa322d77)

The date is derived from the build time.
